server/include/map.h

```diff
@@ -30,13 +35,13 @@ extern "C" {
      * linemate, deraumere, sibur, mendiane, phiras, and thystame.
      */
     typedef struct resources_s {
-        int food_count;
-        int linemate_count;
-        int deraumere_count;
-        int sibur_count;
-        int mendiane_count;
-        int phiras_count;
-        int thystame_count;
+        size_t food_count;
+        size_t linemate_count;
+        size_t deraumere_count;
+        size_t sibur_count;
+        size_t mendiane_count;
+        size_t phiras_count;
+        size_t thystame_count;
     } resources_t;
```

Comment:
You said that this was going to change, it still a structure with a variable for each resource, please make this a map/dictionary the "type" resource should only be as its name stands for uniting resources under a generic name, so create a union.

Reply:
I did it but i remove some hashmap functions that were useless and with too much coding style errors